pkg/utils/nodeclass/nodeclass.go

```diff
@@ -245,3 +251,28 @@ func PatchStatus(ctx context.Context, c client.Client, stored, nodeClass *v1beta
 func HashAnnotation(nodeClass *v1beta1.EC2NodeClass) map[string]string {
 	return map[string]string{v1beta1.AnnotationEC2NodeClassHash: nodeClass.Hash()}
 }
+
+func createSelectorTags(k string, v string, t []map[string]string) []map[string]string {
+	cdlValue := strings.Split(v, ",")
+	for _, val := range cdlValue {
+		for _, tag := range t {
+			if _, ok := tag[k]; ok {
```

Comment:
I think we can assume there are no duplicate values here since the thing that we are propagating from is a map itself.